wave_lang/kernel/wave/analysis/partition_strided_operators.py

```diff
@@ -380,3 +387,117 @@ def has_gpr_offsets(node: fx.Node) -> bool:
                 custom.replace_all_uses_with(reshape)
 
             custom.graph.erase_node(custom.fx_node)
+
+
+def partition_gather_like_ops(trace: CapturedTrace, constraints: list[Constraint]):
+    """
+    This pass partitions gather-like operations (reads/writes with non-contiguous access patterns) into
+    multiple contiguous operations.
+
+    For example, if we have a write operation that writes elements with stride 2:
+    write([0,2,4,6]) -> write([0]), write([2]), write([4]), write([6])
+
+    The pass:
+    1. Identifies reads/writes with non-contiguous access patterns
+    2. For each such operation:
+       - Creates multiple single-element reads/writes
+       - Updates indices to access the correct elements
+       - Handles dynamic values in mappings
+       - Combines results back together if needed (for reads)
+    """
+
+    def has_gather_mapping(node: fx.Node) -> bool:
+        """
+        Checks for writes on 2d tensors with strided access on a single dimension that
+        read more than a single element.
+        """
+        custom = get_custom(node)
+        if isinstance(custom, (Read, Write)):
+            return not custom.is_contiguous_vec()
+
+        return False
+
+    strided_operators = trace.walk(has_gather_mapping)
+    for operator in strided_operators:
+        custom = get_custom(operator)
+        index = custom.index
+        fastest_index = get_fastest_index(index)
+        elements_per_thread = subs_idxc(custom.elements_per_thread)
+
+        # Break apart Reads/Writes that has non-contiguous GPR Read/Writes.
+        with custom.graph.inserting_before(operator):
+            ops_to_combine = []
+            for i in range(elements_per_thread):
+                new_index = deepcopy(index)
+                for j, v in enumerate(new_index.values()):
+                    if j == fastest_index:
+                        v.start = v.start + i
+                    v.size = 1
+                    v.stride = 1
+
+                new_dynamic_vals = []
+                for dynamic_val in custom.mapping_dynamic_vals:
```

Comment:
When you partition an operator, what does that mean for dynamic values that are used by the operator? There are a couple of cases: 1) Single dynamic value in which case you can just use that, 2) N dynamic values where N is the the same as the elements_per_thread and in this case you need to extract the right element and anything else is unclear what we should do. If this is correct, can you add a comment describing that here?